best-practices/haproxy-best-practices.md

```diff
@@ -72,28 +72,51 @@ yum -y install epel-release gcc systemd-devel
 
 ## Deploy HAProxy
 
-You can easily use HAProxy to configure and set up a load-balanced database environment. This section shows general deployment operations. You can customize the [configuration file](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html) based on your actual scenario.
+You can easily use HAProxy to configure and set up a load-balanced database environment. This section shows general deployment operations. You can customize the [configuration file](http://cbonte.github.io/haproxy-dconv/2.5/configuration.html) based on your actual scenario.
 
 ### Install HAProxy
 
-1. Use yum to install HAProxy：
+1. Download the package of the HAProxy 2.5.0 source code:
 
     {{< copyable "shell-regular" >}}
 
     ```bash
-    yum -y install haproxy
+    wget https://github.com/haproxy/haproxy/archive/refs/tags/v2.5.0.zip
     ```
 
-2. Check whether the installation is successful：
+2. Unzip the package:
 
     {{< copyable "shell-regular" >}}
 
     ```bash
-    which haproxy
+    unzip v2.5.0.zip
     ```
 
+3. Compile the application from the source code:
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    cd haproxy-2.5.0
+    make clean
+    make -j 8 TARGET=linux-glibc  USE_THREAD=1
```

Comment:
please remove the redundant space

```
make -j 8 TARGET=linux-glibc USE_THREAD=1
```

Reply:
Thanks! The redundant space is removed in https://github.com/pingcap/docs/pull/7082.